src/WebJobs.Script.Analyzers/WebJobsAttributeAnalyzer.cs

```diff
@@ -0,0 +1,141 @@
+            context.RegisterCompilationStartAction(AnalyzeCompilation);
+        }
+
+        private void AnalyzeCompilation(CompilationStartAnalysisContext context)
+        {
+            var compilation = context.Compilation;
+
+            AssemblyCache.Instance.Build(compilation);
+            _tooling = AssemblyCache.Instance.Tooling;
+
+            // cast to PortableExecutableReference which has a file path
+            var x1 = compilation.References.OfType<PortableExecutableReference>().ToArray();
+            var webJobsPath = (from reference in x1
+                               where IsWebJobsSdk(reference)
+                               select reference.FilePath).SingleOrDefault();
+
+            if (webJobsPath == null)
+            {
+                return; // Not a WebJobs project.
+            }
+        }
+
+        private bool IsWebJobsSdk(PortableExecutableReference reference)
+        {
+            if (reference.FilePath.EndsWith("Microsoft.Azure.WebJobs.dll"))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        // This is called extremely frequently
+        // Analyze the method signature to validate binding attributes + types on the parameters
+        private void AnalyzeMethodDeclarationNode(SyntaxNodeAnalysisContext context)
+        {
+            if (_tooling == null) // Not yet initialized
+            {
+                return;
+            }
+
+            var methodDecl = (MethodDeclarationSyntax)context.Node;
+            var methodName = methodDecl.Identifier.ValueText;
+
+            if (!HasFunctionNameAttribute(context, methodDecl))
+            {
+                return;
+            }
+
+            // TODO: Attribute validation
+        }
+
+        // First argument to the FunctionName ctor.
+        private string GetFunctionNameFromAttribute(SemanticModel semantics, AttributeSyntax attributeSyntax)
+        {
+            foreach (var arg in attributeSyntax.ArgumentList.Arguments)
+            {
+                var val = semantics.GetConstantValue(arg.Expression);
+                if (!val.HasValue)
+                {
+                return null;
+                }
+                return val.Value as string;
+            }
+            return null;
+        }
+
+        // Does the method have a [FunctionName] attribute?
+        // This provides a quick check before we get into the more intensive analysis work.
+        private bool HasFunctionNameAttribute(SyntaxNodeAnalysisContext context, MethodDeclarationSyntax methodDeclarationSyntax)
+        {
+            foreach (var attrListSyntax in methodDeclarationSyntax.AttributeLists)
+            {
+                foreach (AttributeSyntax attributeSyntax in attrListSyntax.Attributes)
+                {
+                    // Perf - Can we get the name without doing a symbol resolution?
```

Comment:
Would `GetText` method give us this info(name) ?

Reply:
`SyntaxNode.GetText` would give a string like `FunctionName("MyFunction")` that the function name could be extracted from. This approach would lose the symbol information, but we're only using the `NamedType.Name` string for comparison. It might be a bit fragile, though.

Reply:
Got it. Thanks!